### PR TITLE
Add `lit` and `FileCheck` to tools and run llvmpasses by default

### PR DESCRIPTION
**Pull request by vchuravy**
Installing `lit` is ugly since it comes as a Python package, but imho still better than pointing it into the source-tree.

#28330 and #28838

### CODE REVIEW

**Review by Keno — APPROVED** [review]
Seems fine to me. Have you checked that everything still works when running the llvmpasses tests with a source build?

File: deps/llvm.mk [hunk 1]
```diff
@@ -461,7 +463,8 @@ endif
 $(build_prefix)/manifest/llvm: | $(llvm_python_workaround)
 
 LLVM_INSTALL = \
-	cd $1 && $$(CMAKE) -DCMAKE_INSTALL_PREFIX="$2$$(build_prefix)" -P cmake_install.cmake
+	cd $1 && cp -r $$(LLVM_SRC_DIR)/utils/lit $2$$(build_depsbindir) && \
```
Comment (vchuravy) [inline deps/llvm.mk]: @vtjnash is there a more elegant way of doing this?

Comment (vtjnash) [reply deps/llvm.mk]: Such as?

**Review by vtjnash — APPROVED** [review]
lgtm

**Comment by vchuravy** [comment]
Now includes #28330 and I fixed a couple of issues that I noted with out-of-tree builds.
LLVMBuilder is ready, LGTM when CI is green.

**Comment by vchuravy** [comment]
Ok fixed a problem with the source install of LLVM. `In order to test this a make -C deps reinstall-llvm` should be sufficient.

**Comment by vchuravy** [comment]
CI failure is network

**Pull request merged**